query/src/main/java/io/keen/client/java/KeenQueryRequest.java

```diff
@@ -0,0 +1,27 @@
+package io.keen.client.java;
+
+import io.keen.client.java.exceptions.KeenQueryClientException;
+import java.net.URL;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Interface to be implemented by a query request
+ *
+ * @author baumatron
+ */
+abstract class KeenQueryRequest {
```

Comment:
This was done due to another Java limitation. I wanted to create a package internal interface. While interfaces themselves can have package-level visibility, methods are forced to have public visibility, so an object implementing an interface would be forced to make all interface methods public outside of the package, which isn't what we want.

Clients shouldn't need to know about any of these methods, probably shouldn't call them and if they do call them it'll have implications for them if we change the internal package object dependencies, something I'd argue is better to keep flexible. It would be better if the methods didn't pop up in auto-complete, too, cluttering things up and possibly making it harder to figure the SDK out. Unfortunately this has the side-effect that an object can only implement one of these "interfaces".